python/python/bystro/proteomics/annotation_interface.py

```diff
@@ -21,15 +22,563 @@
 
 logger = logging.getLogger(__file__)
 
+nest_asyncio.apply()
+
 HETEROZYGOTE_DOSAGE = 1
 HOMOZYGOTE_DOSAGE = 2
-MISSING_GENO_DOSAGE = np.nan
+MISSING_GENO_DOSAGE = -1
 ONE_DAY = "1d"  # default keep_alive time for opensearch point in time index
 
-nest_asyncio.apply()
+CHROM_FIELD = "chrom"
+POS_FIELD = "pos"
+VCF_POS_FIELD = "vcfPos"
+INPUT_REF_FIELD = "inputRef"
+ALT_FIELD = "alt"
+TYPE_FIELD = "type"
+ID_FIELD = "id"
+HETEROZYGOTES_FIELD = "heterozygotes"
+HOMOZYGOTES_FIELD = "homozygotes"
+MISSING_GENOS_FIELD = "missingGenos"
+
+SAMPLE_COLUMNS = [HETEROZYGOTES_FIELD, HOMOZYGOTES_FIELD, MISSING_GENOS_FIELD]
+
+ALWAYS_INCLUDED_FIELDS = [
+    CHROM_FIELD,
+    POS_FIELD,
+    VCF_POS_FIELD,
+    INPUT_REF_FIELD,
+    ALT_FIELD,
+    TYPE_FIELD,
+    ID_FIELD,
+]
+LINK_GENERATED_COLUMN = "locus"
+SAMPLE_GENERATED_COLUMN = "sample"
+DOSAGE_GENERATED_COLUMN = "dosage"
+
+DEFAULT_COLUMN_TYPES = {
+    HETEROZYGOTES_FIELD: str,
+    HOMOZYGOTES_FIELD: str,
+    MISSING_GENOS_FIELD: str,
+    CHROM_FIELD: str,
+    POS_FIELD: np.int64,
+    VCF_POS_FIELD: np.int64,
+    INPUT_REF_FIELD: str,
+    ALT_FIELD: str,
+    TYPE_FIELD: str,
+    ID_FIELD: str,
+    LINK_GENERATED_COLUMN: str,
+    SAMPLE_GENERATED_COLUMN: str,
+    DOSAGE_GENERATED_COLUMN: np.int8,
+}
+
+
+# Fields that may look numeric but are lexical
+DEFAULT_NOT_NUMERIC_FIELDS = [
+    POS_FIELD,
+    VCF_POS_FIELD,
+    ID_FIELD,
+    HETEROZYGOTES_FIELD,
+    HOMOZYGOTES_FIELD,
+    MISSING_GENOS_FIELD,
+    "clinvarVcf.RS",
+    "gnomad.exomes.id",
+    "gnomad.genomes.id",
+    "clinvarVcf.id",
+    "refSeq.codonNumber",
+    "clinvarVcf.ALLELEID",
+    "clinvarVcf.DBVARID",
+    "clinvarVcf.ORIGIN",
+]
+
+DEFAULT_PRIMARY_KEYS = {
+    "gnomad.genomes": "id",
+    "gnomad.exomes": "id",
+    "clinvarVcf": "id",
+    "dbSNP": "id",
+    "nearestTss.refSeq": "name",
+    "nearest.refSeq": "name",
+    "refSeq": "name",
+}
+
+# Tracks that may have more than 1 value per position
+# These we will leave as array-valued, rather than flattening
+DEFAULT_MULTI_VALUED_TRACKS = [
+    "refSeq",
+    "nearest.refSeq",
+    "nearestTss.refSeq",
+]
+
+# The refSeq.clinvar is a legacy track, which has been removed in all recent database versions
+# During join, the refSeq.clinvar track excludes missing values
+# resulting in a reduction in dimension relative to the primary key
+# This libarary assumes that the length of the primary key array
```

Comment:
```suggestion
# This library assumes that the length of the primary key array
```
Found typo, and I'm not sure if this statement is missing something, that the length of the primary key array is ____?